src/rust/ide/lib/view/src/graph_editor/component/project_name.rs

```diff
@@ -0,0 +1,314 @@
+    logger         : Logger,
+    animations     : Animations,
+    view           : component::ShapeView<shape::Shape>,
+    display_object : display::object::Instance,
+    text_field     : TextField,
+    project_name   : Rc<RefCell<String>>,
+    renamed_output : frp::Source<String>,
+}
+
+impl ProjectNameModel {
+    /// Create new ProjectNameModel.
+    pub fn new(world:&World,frp:&Frp) -> Self {
+        let scene                 = world.scene();
+        let logger                = Logger::new("ProjectName");
+        let display_object        = display::object::Instance::new(&logger);
+        let font                  = scene.fonts.get_or_load_embedded_font("DejaVuSansMono").unwrap();
+        let size                  = Vector2::new(600.0,100.0);
+        let base_color            = DARK_GRAY_TEXT_COLOR;
+        let text_size             = 16.0;
+        let text_field_properties = TextFieldProperties{base_color,font,size,text_size};
+        let text_field            = TextField::new(&world,text_field_properties);
+        let view_logger           = Logger::sub(&logger,"view_logger");
+        let view                  = component::ShapeView::<shape::Shape>::new(&view_logger,scene);
+        let project_name          = Rc::new(RefCell::new(UNKNOWN_PROJECT_NAME.to_string()));
+        let renamed_output        = frp.outputs.renamed.clone();
+        let animations            = Animations::new(&frp.network);
+        Self{logger,view,display_object,text_field,project_name,renamed_output
+            ,animations}.initialize()
+    }
+
+    fn setup_center_alignment(&self) {
+        let mut width = 0.0;
+        self.text_field.with_mut_content(|content| {
+            let mut line = content.line(0);
+            if line.len() > 0 {
+                width = line.get_char_x_position(line.len() - 1);
+            }
+        });
+        let offset = Vector3::new(-width/2.0,0.0,0.0);
+        let mut height = default();
+            self.text_field.with_content(|content| height = content.line_height);
+        self.view.shape.sprite.size.set(Vector2::new(width,height));
+        self.view.set_position(Vector3::new(0.0,-height/2.0,0.0));
+        self.animations.positioning.set_target_value(offset);
+    }
+
+    fn initialize(self) -> Self {
+        self.add_child(&self.text_field.display_object());
+        self.add_child(&self.view);
+        let project_name = self.clone_ref();
+        self.text_field.set_text_edit_callback(move |change| {
+            // If the text edit callback is called, the TextEdit must be still alive.
+            let field_content = project_name.text_field.get_content();
+            let new_name      = field_content.replace("\n","");
+            if change.inserted == "\n" {
+                project_name.rename(&new_name);
+            }
+            // Keep only one line.
+            project_name.text_field.set_content(&new_name);
+            project_name.setup_center_alignment();
+        });
```

Comment:
ehh, a lot of logic that is a hack - good its here, but should be removed after new text field is there. Add comment about it pls